website/static/js/comment.js

```diff
@@ -239,34 +367,109 @@ var CommentModel = function(data, $parent, $root) {
         self.unreporting, self.undeleting
     );
 
-    self.isVisible = ko.computed(function() {
-        return !self.isDeleted() && !self.isAbuse();
+    self.isVisible = ko.pureComputed(function() {
+        return !self.isDeleted() && !self.isHidden() && !self.isAbuse();
     });
 
-    self.editNotEmpty = ko.computed(function() {
+    self.editNotEmpty = ko.pureComputed(function() {
         return notEmpty(self.content());
     });
 
-    self.toggleIcon = ko.computed(function() {
-            return self.showChildren() ? 'fa fa-minus-square-o' : 'fa fa-plus-square-o';
+    self.toggleIcon = ko.pureComputed(function() {
+        return self.showChildren() ? 'fa fa-minus-square-o' : 'fa fa-plus-square-o';
     });
-    self.editHighlight = ko.computed(function() {
-        return self.canEdit() && self.hoverContent();
+    self.editHighlight = ko.pureComputed(function() {
+        return self.canEdit() && self.hoverContent() && self.mode !== 'widget';
     });
-    self.canReport = ko.computed(function() {
+    self.canReport = ko.pureComputed(function() {
         return self.$root.canComment() && !self.canEdit();
     });
 
-    self.shouldShow = ko.computed(function() {
-        return !self.isDeleted() || self.hasChildren() || self.canEdit();
+    self.shouldShow = ko.pureComputed(function() {
+        if (!self.isDeleted() && !self.isHidden()) {
+            return true;
+        }
+        if (self.isHidden()) {
+            return self.level === 1;
+        }
+        return self.hasChildren() || self.canEdit();
+    });
+
+    self.shouldShowChildren = ko.computed(function() {
+        if (self.isHidden()) {
+            self.showChildren(false);
+            return false;
+        }
+        return self.level < self.MAXLEVEL;
+    });
+
+    self.shouldContinueThread = ko.pureComputed(function() {
+        if (self.shouldShowChildren()) { return false;}
+        return ((!self.isHidden()) && self.hasChildren());
+    })
+
+    self.cleanTitle = ko.pureComputed(function() {
+        var cleaned;
+        switch(self.page()) {
+            case 'wiki':
+                cleaned = '(Wiki';
+                if (self.title().toLowerCase() !== 'home') {
+                    cleaned += ' - ' + self.title();
+                }
+                break;
+            case 'files':
+                cleaned = '(Files - ';
+                var path = self.title().split('/');
+                cleaned += path[path.length - 1];
+                break;
+            case 'node':
+                cleaned = '(Overview';
+                break;
+        }
+        cleaned += ')';
+        return unescape(cleaned);
```

Comment:
Looks like `unescape()` is a depricated js method: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape. I'll use `decodeURIComponent` instead.